### PR TITLE
[ext] Condense `dagster-ext` package to one file

### PR DESCRIPTION
**Pull request by smackesey**
## Summary & Motivation

Condenses `dagster-ext` into a single file `dagster_ext/__init__.py` so it can be vendored easily.

I didn't go all the way to just `dagster_ext.py` so that I could leave `dagster_ext/version.py` and `dagster_ext/py.typed`. It should be just as easy to copy-paste the file contents from `__init__.py`.

## How I Tested These Changes

Existing test suite.

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #16321** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16321" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/16321?utm_source=stack-comment).

File: python_modules/dagster-ext/dagster_ext/__init__.py [hunk 1]
```diff
@@ -1,32 +1,732 @@
...
+# ExtExtras as ExtExtras,
+# ExtMessage as ExtMessage,
+# ExtParams as ExtParams,
+# ExtPartitionKeyRange as ExtPartitionKeyRange,
+# ExtTimeWindow as ExtTimeWindow,
+# DagsterExtError as DagsterExtError,
+# DagsterExtWarning as DagsterExtWarning,
+# decode_env_var as decode_env_var,
+# encode_env_var as encode_env_var,
+# is_dagster_orchestration_active as is_dagster_orchestration_active,
```
Comment (schrockn) [inline python_modules/dagster-ext/dagster_ext/__init__.py:732]: hmmm very skeptical of the utility of this. almost certainly will fall out of date

**Review by schrockn — CHANGES_REQUESTED** [review]
not sure about that public API section. Seems like it will just become out of date without any enforcement

**Review by schrockn — APPROVED** [review]

File: python_modules/dagster-ext/dagster_ext/__init__.py (hunk 1 above)
Comment (smackesey) [reply python_modules/dagster-ext/dagster_ext/__init__.py]: fair, it was more useful while I was picking out the methods to underscore than it is now, removed

**Pull request merged**